src/Component/InlineEdit.php

```diff
@@ -0,0 +1,179 @@
+<?php
+/**
+ * A Simple inline editable text Vue component.
+ */
+
+namespace atk4\ui\Component;
+
+use atk4\ui\Exception;
+use atk4\ui\jsVueService;
+use atk4\ui\View;
+
+class InlineEdit extends View
+{
+    public $defaultTemplate = 'inline-edit.html';
+
+    /**
+     * jsCallback for saving data.
+     *
+     * @var null
+     */
+    public $cb = null;
+
+    /**
+     * Whether callback should save value to db automatically or not.
+     * Default to using onChange handler.
+     * If set to true, then saving to db will be done when model get set
+     * and if model is loaded already.
+     *
+     * @var bool
+     */
+    public $autoSave = false;
+
+    /**
+     * The actual db field name that need to be saved.
+     * Default to title field when model is set.
+     *
+     * @var null|string The name of the field.
+     */
+    public $modelField = null;
+
+    /**
+     * Whether component should save it's value when input get blur.
+     *
+     * @var bool
+     */
+    public $saveOnBlur = true;
+
+    /**
+     * Default css for the input div.
+     *
+     * @var string
+     */
+    public $inputCss = 'ui right icon input';
+
+    /**
+     * Initialization.
+     */
+    public function init()
+    {
+        parent::init();
+        $this->cb = $this->add('jsCallback');
+    }
+
+    /**
+     * Set Model of this View.
+     *
+     * @param \atk4\data\Model $model
+     *
+     * @return \atk4\data\Model
+     */
+    public function setModel(\atk4\data\Model $model)
+    {
+        parent::setModel($model);
+        $this->modelField = $this->modelField ? $this->modelField : $this->model->title_field;
+        if ($this->autoSave && $this->model->loaded()) {
+            if ($this->cb->triggered()) {
+                $value = $_POST['value'] ? $_POST['value'] : null;
```

Comment:
it is possible for a callback to pass value. See https://github.com/atk4/ui/blob/develop/docs/callbacks.rst and argument to "getURL()".

so the next line would be:

$this->cb->set(function($value) {